Formula/llvm-arm-embedded.rb

```diff
@@ -0,0 +1,102 @@
+class LlvmArmEmbedded < Formula
+  desc "LLVM toolchain for Arm embedded targets"
+  homepage "https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm"
+  license all_of: [
+    { "Apache-2.0" => { with: "LLVM-exception" } },
+    "BSD-3-Clause",
+  ]
+
+  # build process requires git information, so the repos have to be downloaded in full
+  stable do
+    url "https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm.git",
+        tag:    "release-16.0.0",
+        commit: "c1dbc6bf94ab1009400ec430ff0dd7f921044cce"
+
+    resource "llvm" do
+      url "https://github.com/llvm/llvm-project.git",
+          tag:    "llvmorg-16.0.3",
+          commit: "da3cd333bea572fb10470f610a27f22bcb84b08c"
+    end
+
+    resource "picolibc" do
+      url "https://github.com/picolibc/picolibc.git",
+          tag:    "1.8.1",
+          commit: "93b5d5f2ad44867b60267417cd6d6250dbf68983"
+    end
+  end
+
+  head do
+    url "https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm.git",
+        branch: "main"
+
+    resource "llvm" do
+      url "https://github.com/llvm/llvm-project.git",
+          branch: "main"
+    end
+
+    resource "picolibc" do
+      url "https://github.com/picolibc/picolibc.git",
+          branch: "main"
+    end
+  end
+
+  keg_only "this is a specialized build of LLVM that would conflict with the standard build"
+
+  depends_on "cmake" => :build
+  depends_on "gnu-tar" => :build
+  depends_on "gpatch" => :build
```

Comment:
We can assume that `tar` and `patch` are available; they are used by `brew` too. (Unless macOS's BSD versions don't work here.)

Reply:
They should!  Can we assume the same on Linux?

Reply:
Yes. For both platforms, we can assume that these system-provided tools are sufficient, unless proven otherwise.